include/z64player.h

```diff
@@ -431,7 +431,7 @@ typedef struct {
 #define PLAYER_STATE2_12 (1 << 12)
 #define PLAYER_STATE2_13 (1 << 13)
 #define PLAYER_STATE2_14 (1 << 14)
-#define PLAYER_STATE2_15 (1 << 15)
+#define PLAYER_STATE2_FREEZE_WITH_GRAVITY (1 << 15) // Replaces a large block in the middle of Player_UpdateCommon with only gravity and bgCheck updating (only used by Twinrova's freezing effects)
```

Comment:
Only twinrova ends up using this yeah, but there's nothing twinrova specific about it. Can think of many other places someone would want to use this (it should honestly be used in most cases where actor.freezTtimer is used)

Reply:
I mean, for now we don't know exactly what this skips because this chunk of code is huge and player is mostly undocumented, and also I have the impression this is a Morithack

Reply:
You dont really need to know the details of what the large update block does, cause all that matters is what its replaced with. It can be boiled down to: 
- stop horizontal movement
- call move forward (allow falling to the ground if airborne. no horiz movement cause of the above)
- update bgcheck info so he doesnt fall through the floor.

This is objectively a better method of freezing link because it allows some important aspects of his actor to update even tho the "main logic" will not. If this was used instead of freezeTimer in some instances it would even prevent SRM and related bugs from being able to manifest.

I dont doubt it was added by morita for his use case, but it doesnt mean it has to be named after that as if it only makes sense in that context. Its pretty general and useful outside of that.

Reply:
Not to mention this state/code exists in mm and is used by multiple things. its broader than just a twinrova thing